### PR TITLE
Implement the DocGenerator main module and test search

### PR DESCRIPTION
**Pull request by palaciosjeremias**
|Related issue|
|---|
|#1606|

## Description

This PR implements the main module of Doc Generator.
This module is in charge of searching for every test based on the configuration provided and calls the CodeParser for each of them.
Also, it searches for Group files (by default README files) and calls the CodeParser for them. 
For each test found, a JSON and a YAML files are created with the parsed information. For each README file, a JSON and a YAML files are created with the group description.
The ID and the group ID are created and handed to the CodeParser.

### CODE REVIEW

File: docs/DocGenerator/Config.py [hunk 1]
```diff
@@ -10,6 +10,8 @@ def __init__(self):
         self.documentation_path = ".."
```
Comment (MiguelazoDS) [inline docs/DocGenerator/Config.py:10]: If you want to keep this file (with capital C) remove the other one (config.py)

File: docs/DocGenerator/config.yaml [hunk 2]
```diff
@@ -10,11 +10,17 @@ Include paths:
...
+Group files:
+  - "README.md"
+
+Function regex:
+  - "^test_"
 
 Ignore paths:
   - "../../tests/integration/test_wazuh_db/data"
-  - "*/data/*"
+  - "/data/*"
```
Comment (MiguelazoDS) [inline docs/DocGenerator/config.yaml:23]: Are you checking the directories level by level? How is exactly used this regex?

File: docs/DocGenerator/DocGenerator.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        doc_path = os.path.splitext(base_path + relative_path)[0]
+        return doc_path
+
+    def create_group(self, path, group_id):
+        self.__id_counter = self.__id_counter + 1
+        group = self.parser.parse_group(path, self.__id_counter, group_id)
+        doc_path = self.get_group_doc_path(path)
+        if not os.path.exists(os.path.dirname(doc_path)):
+            os.makedirs(os.path.dirname(doc_path))
+        with open(doc_path + ".json", "w+") as outfile:
```
Comment (MiguelazoDS) [inline docs/DocGenerator/DocGenerator.py:66]: Would not be better a method to avoid repeat that?

File: docs/DocGenerator/DocGenerator.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+            outfile.write(yaml.dump(group))
+        return self.__id_counter
+
+    def create_test(self, path, group_id):
+        self.__id_counter = self.__id_counter + 1
+        test = self.parser.parse_test(path, self.__id_counter, group_id)
+        doc_path = self.get_test_doc_path(path)
+        if not os.path.exists( os.path.dirname(doc_path) ):
+            os.makedirs(os.path.dirname(doc_path))
+        with open(doc_path + ".json", "w+") as outfile:
```
Comment (MiguelazoDS) [inline docs/DocGenerator/DocGenerator.py:78]: Same here

File: docs/DocGenerator/config.yaml (hunk 2 above)
Comment (palaciosjeremias) [reply docs/DocGenerator/config.yaml:23]: These 3 options are examples of how the Ignore path can be implemented. All of them have different levels of impact.

File: docs/DocGenerator/Config.py (hunk 1 above)
Comment (palaciosjeremias) [reply docs/DocGenerator/Config.py:10]: Fixed!

File: docs/DocGenerator/DocGenerator.py (hunk 3 above)
Comment (palaciosjeremias) [reply docs/DocGenerator/DocGenerator.py:63]: You are right. It´s improved now.

File: docs/DocGenerator/DocGenerator.py (hunk 4 above)
Comment (palaciosjeremias) [reply docs/DocGenerator/DocGenerator.py]: Improved!

**Review by MiguelazoDS — APPROVED** [review]

**Pull request merged**